services/brig/src/Brig/Run.hs

```diff
@@ -103,3 +124,51 @@ lookupRequestIdMiddleware :: (RequestId -> Wai.Application) -> Wai.Application
 lookupRequestIdMiddleware mkapp req cont = do
   let reqid = maybe def RequestId $ lookupRequestId req
   mkapp reqid req cont
+
+cleanExpiredPendingInvitations :: AppIO ()
+cleanExpiredPendingInvitations = do
+  safeForever "cleanExpiredPendingInvitations" $ do
+    today <- utctDay <$> (liftIO =<< view currentTime)
+    for_ [0 .. 9] $ \i ->
+      cleanUpDay (addDays (- i) today)
+    let d :: Int = 24 * 60 * 60
+    randomSecs <- liftIO (round <$> randomRIO @Double (0.5 * fromIntegral d, fromIntegral d))
+    threadDelay (randomSecs * 1_000_000)
+  where
+    cleanUpDay :: Day -> AppIO ()
+    cleanUpDay day = do
+      expiredEntries <- forTrackedExpirations day $ \exps ->
+        catMaybes
+          <$> ( for exps $ \(PendingActivationExpiration uid expiresAt tid) -> do
+                  isExpired <- (expiresAt <=) <$> (liftIO =<< view currentTime)
```

Comment:
As I wrote in jira: we can simplify this logic and remove the `expiresAt` field altogether, at the cost of having data dangle for (at most) 24h longer: just process full days, but start with things that expired *some time* yesterday, not today.  I think this will also elminate the need for the `Ord` instance.

Reply:
Ah yes, right. Done